editor/doc_tools.cpp

```diff
@@ -76,241 +76,275 @@ static String _translate_doc_string(const String &p_text) {
+	}
+};
 
-		const DocData::ClassDoc &cf = p_data.class_list[c.name];
+// Comparator for methods, compares on names.
+struct MethodCmp {
+	_FORCE_INLINE_ bool operator()(const DocData::MethodDoc &p_lhs, const DocData::MethodDoc &p_rhs) const {
+		return p_lhs.name.naturalcasecmp_to(p_rhs.name) < 0;
+	}
+};
 
-		c.is_deprecated = cf.is_deprecated;
-		c.deprecated_message = cf.deprecated_message;
-		c.is_experimental = cf.is_experimental;
-		c.experimental_message = cf.experimental_message;
-		c.keywords = cf.keywords;
+static void merge_constructors(Vector<DocData::MethodDoc> &p_to, const Vector<DocData::MethodDoc> &p_from) {
+	// Get data from `p_from`, to avoid mutation checks.
+	const DocData::MethodDoc *from_ptr = p_from.ptr();
+	int64_t from_size = p_from.size();
 
-		c.description = cf.description;
-		c.brief_description = cf.brief_description;
-		c.tutorials = cf.tutorials;
+	// TODO: Improve constructor merging.
+	for (DocData::MethodDoc &to : p_to) {
+		for (int64_t from_i = 0; from_i < from_size; ++from_i) {
+			const DocData::MethodDoc &from = from_ptr[from_i];
 
-		for (int i = 0; i < c.constructors.size(); i++) {
-			DocData::MethodDoc &m = c.constructors.write[i];
+			// Compare argument count first.
+			if (from.arguments.size() != to.arguments.size()) {
+				continue;
+			}
 
-			for (int j = 0; j < cf.constructors.size(); j++) {
-				if (cf.constructors[j].name != m.name) {
-					continue;
-				}
+			if (from.name != to.name) {
+				continue;
+			}
 
-				{
-					// Since constructors can repeat, we need to check the type of
-					// the arguments so we make sure they are different.
-					if (cf.constructors[j].arguments.size() != m.arguments.size()) {
-						continue;
-					}
-					int arg_count = cf.constructors[j].arguments.size();
-					Vector<bool> arg_used;
-					arg_used.resize(arg_count);
-					for (int l = 0; l < arg_count; ++l) {
-						arg_used.write[l] = false;
-					}
-					// also there is no guarantee that argument ordering will match, so we
-					// have to check one by one so we make sure we have an exact match
-					for (int k = 0; k < arg_count; ++k) {
-						for (int l = 0; l < arg_count; ++l) {
-							if (cf.constructors[j].arguments[k].type == m.arguments[l].type && !arg_used[l]) {
-								arg_used.write[l] = true;
-								break;
-							}
-						}
-					}
-					bool not_the_same = false;
-					for (int l = 0; l < arg_count; ++l) {
-						if (!arg_used[l]) { // at least one of the arguments was different
-							not_the_same = true;
+			{
+				// Since constructors can repeat, we need to check the type of
+				// the arguments so we make sure they are different.
+				int64_t arg_count = from.arguments.size();
+				Vector<bool> arg_used;
+				arg_used.resize_zeroed(arg_count);
+				// Also there is no guarantee that argument ordering will match,
+				// so we have to check one by one so we make sure we have an exact match.
+				for (int64_t arg_i = 0; arg_i < arg_count; ++arg_i) {
```

Comment:
is `++arg_i` as opposed to `arg_i++` important here? If not, I'd think you'd want to use `arg_i++` just to be consistent with the rest of the code.

Reply:
This is just the original code

Reply:
Maybe I'm misreading the git diff, but I'm seeing lots of red lines that have e.g. `i++` and `j++` in them, so that's what I what I was referring to for being consistent.

Reply:
But that's consistent with code that doesn't exist any longer :) The code that *is* in the result is consistent

With simple types it doesn't matter if you do pre- or postincrement, but for the iterator it's not necessarily trivial, which is why I used it originally, but it doesn't matter here, and it's far away from any other code to be consistent with, there's no code specification for which is preferred (but preincrement is never going to be worse performance wise than postincrement, when it does matter)